lib/buffer.js

```diff
@@ -67,10 +73,67 @@ function Buffer(arg, encoding) {
   return fromObject(arg);
 }
 
+/**
+ * Functionally equivalent to Buffer(arg, encoding) but throws a TypeError
+ * if value is a number.
+ **/
+Buffer.from = function(value, encoding) {
+  if (typeof value === 'number')
+    throw new TypeError(`'value' argument must not be a number`);
+  return typeof value === 'string' ?
+    fromString(value, encoding) :
+    fromObject(value);
+};
+
 Object.setPrototypeOf(Buffer.prototype, Uint8Array.prototype);
 Object.setPrototypeOf(Buffer, Uint8Array);
 
+/**
+ * Creates a new filled Buffer instance.
+ **/
+Buffer.alloc = function(size, fill, encoding) {
+  if (typeof size !== 'number')
+    throw new TypeError(`'size' argument must be a number`);
+  size = Math.max(0, size);
+  if (typeof fill !== 'undefined') {
+    if (encoding) {
+      if (typeof encoding !== 'string')
+        throw new TypeError(`'encoding' argument must be a string`);
+      if (typeof fill !== 'string')
+        throw new TypeError(
+          `If 'encoding' is a string, 'fill' must also be a string`);
```

Comment:
Current `Buffer#fill()` behavior does accept strings. Should probably mirror that behavior. May just want to punt on the error throw to `fill()` since it already does all necessary checks.